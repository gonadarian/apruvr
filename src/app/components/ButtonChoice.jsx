/* @flow */
import React, { type Element } from 'react';
import { map, pick } from 'lodash';
import Button from './Button';

type Props = {|
    choices: {[id: string]: boolean},
    used: string[],
    names: {[id: string]: string},
    onChoose: (key: string) => void,
|};

export const ButtonChoice = ({ choices, used, names, onChoose }: Props): Element<*> =>
    <div className="btn-group">
        {map(
            pick(choices, used),
            (choice, key): Element<*> =>
                <Button
                    key={key}
                    name={names[key]}
                    isActive={choice}
                    onClick={(): void => onChoose(key)} />
        )}
    </div>;

export default ButtonChoice;
