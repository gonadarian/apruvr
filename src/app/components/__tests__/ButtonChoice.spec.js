import React from 'react';
import renderer from 'react-test-renderer';
import { ButtonChoice } from '../ButtonChoice';

describe('ButtonChoice component', () => {
    /* eslint-disable no-magic-numbers */

    it('renders correctly by default', () => {
        const tree = renderer.create(
            <ButtonChoice />
        ).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.type).toEqual('div');
        expect(tree.props.children).toBeUndefined();
    });

    it('renders correctly with single choice', () => {
        const tree = renderer.create(
            <ButtonChoice
                choices={{ 'work': true }}
                used={['work']}
                names={{ 'work': 'Work' }} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.children).toHaveLength(1);
        expect(tree.children[0].props.className).toEqual(expect.stringContaining('primary'));
        expect(tree.children[0].children).toContain('Work');
    });

    it('renders correctly with multiple choices', () => {
        const tree = renderer.create(
            <ButtonChoice
                choices={{ 'work': false, 'sleep': true }}
                used={['work', 'sleep']}
                names={{ 'work': 'Work', 'sleep': 'Sleep' }} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.children).toHaveLength(2);
        expect(tree.children[0].props.className).toEqual(expect.stringContaining('default'));
        expect(tree.children[1].props.className).toEqual(expect.stringContaining('primary'));
        expect(tree.children[1].children).toContain('Sleep');
    });

});
