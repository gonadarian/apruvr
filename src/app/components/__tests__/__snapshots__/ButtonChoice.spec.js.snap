// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonChoice component renders correctly by default 1`] = `
<div
  className="btn-group"
/>
`;

exports[`ButtonChoice component renders correctly with multiple choices 1`] = `
<div
  className="btn-group"
>
  <div
    className="btn btn-default"
    onClick={[Function]}
  >
    Work
  </div>
  <div
    className="btn btn-primary"
    onClick={[Function]}
  >
    Sleep
  </div>
</div>
`;

exports[`ButtonChoice component renders correctly with single choice 1`] = `
<div
  className="btn-group"
>
  <div
    className="btn btn-primary"
    onClick={[Function]}
  >
    Work
  </div>
</div>
`;
